### PR TITLE
Update SDL-0183 (Automatic Mobile Resumption Hash Management) with requested revisions

### PR DESCRIPTION
**Pull request by joeljfischer**
## Updates
* Rework to use an approach of automatically enabling the reuse of data set by the developer if it was resumed.
* Update alternatives with a new alternative (3)
* Add *previous approaches* section

### CODE REVIEW

**Comment by joeljfischer** [comment]
@theresalech This is ready for review

File: proposals/0183-mobile-hash-managment.md [hunk 1]
```diff
@@ -7,53 +7,71 @@
 
 ## Introduction
 
-Add automatic hash management to the manager layer of the libraries. Automatically send the resumption hash upon connection and refill the managers with the data they contained at the time that hash was retrieved.
+Add automatic hash management to the manager layer of the libraries. Automatically send the resumption hash upon connection and enable the reuse of data the developer sets if it already exists.
 
 ## Motivation
 
-With the addition of the manager layer, hashes have become significantly more difficult to use. While hashes can be sent and the data will resume on the system, the manager-layer does not know this and cannot restore the data that is currently in use. This may cause bugs or for data to be resent in its entirety that merely needs to be updated.
+With the addition of the manager layer, hashes have become significantly more difficult to use. While hashes can be sent and the data will resumed on the system, the manager-layer does not know this and cannot restore the data that is currently in use. This may cause bugs or for data to be resent in its entirety that merely needs to be reused.
```
Comment (theresalech) [inline proposals/0183-mobile-hash-managment.md]: I think the second sentence needs to be updated: "...and the data will be resumed on..."

File: proposals/0183-mobile-hash-managment.md [hunk 2]
```diff
@@ -7,53 +7,71 @@
...
 
 ## Proposed solution
 
-The general proposed solution is to store similar sorts of data that the head unit stores for a hash. When the hash is accepted, the library should auto-restore data into the various managers as well. This will allow the developer to interact with the managers as desired.
+The general proposed solution is to store similar sorts of data that the head unit stores for a hash. When the hash is accepted, data will not be set onto the managers. Instead, the data will be held in a "waiting" state on the new `SDLResumptionManager` manager, though it should be inspectable by the developer. When the developer sets up their menus and soft buttons, if those objects exactly match what has been stored, no RPCs will be sent, but callbacks will be setup for those objects.
 
-This change would affect the sub-screen managers only: the Choice Set, Menu, Text and Graphic, and Soft Button Managers.
+A new delegate method will be added to `SDLManagerDelegate` and it's Android equivalent.
```
Comment (theresalech) [inline proposals/0183-mobile-hash-managment.md]: Typo: "its"

**Comment by joeljfischer** [comment]
@theresalech fixed!

**Pull request merged**